contributors/design-proposals/federated-resource-quota.md

```diff
@@ -0,0 +1,53 @@
+# Federated ResourceQuota Requirements & Design
+@weijinxu, @IrfanUrRehman, @JianhuiZ
+reviewed by  @quinton-hoole, @deepak-vijSummary
+
+In a single Kubernetes cluster there is a concept of Resource Quota that manages the resources by namespace so that user usages can be limited to their fair share of the resources. We want to expand this concept to the federation level so that users can treat Federation Control Plane as a “Single Pane of Glass” for seamlessly managing the Resource Quota  information across all the underlying federated clusters.
+
+## User scenario
+
+As a federation user and an admin of clusters, I want to be able to define an overall resource quota for some specific or all resources for a given federated namespace. The overall resource quota limit should be applicable across clusters and preferably enforced at federation control plane level.
+
+For example, let us say that we’ve specified 100 pods in the resource quota spec in the federation namespace, with multiple underlying federated clusters. Let us assume that 50 pods have been deployed already across the federation in a previous deployment run. In a subsequent deployment run, if a user requests additional 50+ more pods to be deployed, the deployment will be rejected as a whole by the federation control plane before deployment process actually starts. If user requests less than 50 pods, the deployment should succeed as per the deployment preferences.
+
+## Implementation details
+### Create Local Resource Quota in underlying clusters
+
+A straightforward thought is to divide and propagate the federation quota value to the underlying clusters. This kind of distribution may cause divided quota gets too small share which hardly meet users requests, and may end up in a state that a user may have quite a few resources available but not in the cluster he/she preferred. What makes this worse is that the user wants to deploy a big workload but none of the clusters have enough resource quota value left anymore. This means that we may need to somehow adjust the allocated quota of the underlying clusters accordingly. Similarly, this is also true for clusters joining and leaving the federation.
+Our proposal is that to not split the quota value but instead give every underlying cluster the entire federation quota. When a user creates Federation Resource Quota, Federation Resource Quota controller will propagate the federation resource quota value to each federated cluster. At the same time, the Federation Resource Quota controller then keeps monitoring underlying Local Resource Quota information and updates the Federation Resource Quota status with already used/hard resource information to be subsequently used by the Federation Resource Quota Admission Controller.
```

Comment:
@weijinxu, This detail of this proposal should be explicit with some as follows:

The fcp will create and monitor the ***resource quota objects*** locally into each federated clusters, initially with the same quota value as submitted to fcp. The ***status*** (actual run time usage) of the quota will be updated against these local quota object by the cluster local quota controller(s) in one or more clusters, which fcp will use to update the quota value (spec) against individual quota objects across the federated clusters.

You have used ***resource quota value*** at most of the places, which makes the method of enforcement unclear. Please follow this scheme of usage across the doc including the details in the examples.